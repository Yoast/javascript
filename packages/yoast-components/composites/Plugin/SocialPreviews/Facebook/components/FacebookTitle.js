import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const FacebookTitleWrapper = styled.span`
	color: #1d2129;
	font-weight: 600;
	overflow: hidden;
	font-size: 16px;
	line-height: 20px;
	margin: 5px 0 0;
	max-height: 110px;
	word-wrap: break-word;
	letter-spacing: normal;
	text-overflow: ellipsis;
	white-space: normal;
	cursor: pointer;
`;

/**
 * Renders a FacebookTitle component.
 *
 * @param {object} props The props.
 *
 * @returns {React.Element} The rendered element.
 */
const FacebookTitle = ( props ) => {
	return (
		<FacebookTitleWrapper>
			{ props.title }
		</FacebookTitleWrapper>
	);
};

FacebookTitle.propTypes = {
	title: PropTypes.string.isRequired,
};

export default FacebookTitle;
