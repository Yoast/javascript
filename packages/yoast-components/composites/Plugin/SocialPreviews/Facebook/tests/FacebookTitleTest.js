/* External dependencies */
import React from "react";
import renderer from "react-test-renderer";

/* Internal dependencies */
import FacebookTitle from "../components/FacebookTitle";

describe( "FacebookTitle", () => {
	it( "matches the snapshot by default", () => {
		const component = renderer.create(
			<FacebookTitle title="YoastCon Workshops" />
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
