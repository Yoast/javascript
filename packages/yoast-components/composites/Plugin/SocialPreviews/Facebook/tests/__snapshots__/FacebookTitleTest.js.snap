// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookTitle matches the snapshot by default 1`] = `
.c0 {
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  line-height: 20px;
  margin: 5px 0 0;
  max-height: 110px;
  word-wrap: break-word;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-overflow: ellipsis;
  white-space: normal;
  cursor: pointer;
}

<span
  className="c0"
>
  YoastCon Workshops
</span>
`;
